295 create new route path geometry select node

Divinare opened the pull request:
closes #295 

About this PR

* package.json:
   * added @types/graphql cause yarn start needed it after removing node_modules
   * added custom leaflet.vectorgrid because version 1.3.0 has a bug on click event handling. Added a card to get back to original vectorgrid after this bug is solved: https://github.com/HSLdevcom/jore-map-ui/issues/335

* clicked node is fetched and it's properties id and coordinates are taken from that response. I didn't use clickedTarget's properties because of jore gives those coordinates in "3 different ways" (and I want to be sure to get the correct lat lon):

solmx:"60.238219"
solmy:"24.904694"

solomx:"60.244068"
solomy:"24.896084"

solstmx:"60.244031"
solstmy:"24.896164"

* changed a bit of coloring of network nodes. Had to make them bigger for easier clicking

* Part 2 where this implementation continues: https://github.com/HSLdevcom/jore-map-ui/issues/296

kronken commented on yarn.lock:
```diff
@@ -6299,30 +7188,36 @@ react-dev-utils@^5.0.1:
-  resolved "https://registry.yarnpkg.com/react-icons/-/react-icons-3.0.5.tgz#9f22a319a48b562e052af2255a3dda8ded363d1c"
+  version "3.2.1"
+  resolved "https://registry.yarnpkg.com/react-icons/-/react-icons-3.2.1.tgz#6f7f9ccbeec05ff7076df070703e03238903ac30"
+  integrity sha512-NQhnh7G5hWCAK7pXPeeI944RkeZ85mtvFATx3wfcVQRTW4ejaQgriemYF72gQIFIyoApMXLTdb01BKp8PaXlrg==
```
This update breaks icons

kronken commented on src/components/map/layers/NetworkLayers.tsx:
```diff
@@ -67,28 +77,55 @@ export default class NetworkLayers extends Component<INetworkLayersProps> {
 
+    private addNodeFromClickEvent = (clickEvent: any) => {
+        const properties =  clickEvent.sourceTarget.properties;
+        NodeService.fetchNode(properties.soltunnus).then((node) => {
```
Saw your comment on the PR

kronken replied:
Would it be possible to run the target.properties through the factory? By doing that we could ensure that the coordinates we store, are the same that are used in other places. But, still would not require fetching it

Divinare replied:
Added comment

kronken commented on yarn.lock:
```diff
@@ -4483,139 +5066,161 @@ jsdom@^11.5.1:
+"leaflet.vectorgrid@git+https://github.com/chriszrc/Leaflet.VectorGrid.git":
   version "1.3.0"
-  resolved "https://registry.yarnpkg.com/leaflet.vectorgrid/-/leaflet.vectorgrid-1.3.0.tgz#ff2deea17d06a9c6cc5e65d8c7287093950e219a"
+  resolved "git+https://github.com/chriszrc/Leaflet.VectorGrid.git#6bdad6d07290db04ff02ebf44830b85bcfe6c020"
```
I assume it is this that breaks the travis run. In other projects we use a syntax like this, to reference to github repos:

"hsl-map-style@github:hsldevcom/hsl-map-style#c808370112717921eeceeb34af1335de2e21daab":
     version "1.0.1"
     resolved "https://codeload.github.com/HSLdevcom/hsl-map-style/tar.gz/c808370112717921eeceeb34af1335de2e21daab"
    dependencies:
        lodash "^4.17.4"

Divinare commented:
Review fixes done

Divinare commented:
About my review fixes:
* Fixed getters / actions order at stores, first list all actions and then getters
   * Also getters / actions should be imo listed in the same order as they are declared

* Added some TODO comments to mapStore. Card: https://github.com/HSLdevcom/jore-map-ui/issues/336
